configstack/module.go

```diff
@@ -57,26 +59,85 @@ func ResolveTerraformModules(terragruntConfigPaths []string, terragruntOptions *
 		return nil, err
 	}
 
-	modules, err := resolveModules(canonicalTerragruntConfigPaths, terragruntOptions, childTerragruntConfig, howThesePathsWereFound)
+	var modules map[string]*TerraformModule
+	err = telemetry.Telemetry(terragruntOptions, "resolve_modules", map[string]interface{}{
+		"working_dir": terragruntOptions.WorkingDir,
+	}, func(childCtx context.Context) error {
+
+		result, err := resolveModules(canonicalTerragruntConfigPaths, terragruntOptions, childTerragruntConfig, howThesePathsWereFound)
+		if err != nil {
+			return err
+		}
+		modules = result
+		return nil
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	externalDependencies, err := resolveExternalDependenciesForModules(modules, map[string]*TerraformModule{}, 0, terragruntOptions, childTerragruntConfig)
+	var externalDependencies map[string]*TerraformModule
+	err = telemetry.Telemetry(terragruntOptions, "resolve_external_dependencies_for_modules", map[string]interface{}{
+		"working_dir": terragruntOptions.WorkingDir,
+	}, func(childCtx context.Context) error {
+		result, err := resolveExternalDependenciesForModules(modules, map[string]*TerraformModule{}, 0, terragruntOptions, childTerragruntConfig)
+		if err != nil {
+			return err
+		}
+		externalDependencies = result
+		return nil
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	crossLinkedModules, err := crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+	var crossLinkedModules []*TerraformModule
+	err = telemetry.Telemetry(terragruntOptions, "crosslink_dependencies", map[string]interface{}{
+		"working_dir": terragruntOptions.WorkingDir,
+	}, func(childCtx context.Context) error {
+		result, err := crosslinkDependencies(mergeMaps(modules, externalDependencies), canonicalTerragruntConfigPaths)
+		if err != nil {
+			return err
+		}
+		crossLinkedModules = result
+		return nil
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	includedModules := flagIncludedDirs(crossLinkedModules, terragruntOptions)
+	var includedModules []*TerraformModule
+	err = telemetry.Telemetry(terragruntOptions, "flag_included_dirs", map[string]interface{}{
```

Comment:
Good to see lots of extra telemetry 👍 

Was it useful in tracking down these perf issues?

Reply:
Yes, it was helpful to find which part consumes more time during the execution